spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -185,6 +186,14 @@ public RewriteDataFiles.Result execute() {
     }
   }
 
+  @Override
+  protected SparkSession spark() {
```

Comment:
This may only be called once right now, but I think we should not assume that it always will be. Can you update this to keep a copy of the session for this action?

Reply:
I think what he meant is that, have a local variable of `SparkSession`  in `RewriteDataFilesSparkAction`, initialize it once by cloning the session and disabling adaptive encoding. 

Whenever spark() is called, return that variable instead of cloning again and again.

Reply:
Good point. It looks there's no need for overriding this method but simply cloning in the constructor of `RewriteDataFilesSparkAction`